src/framing/plaintext.rs

```diff
@@ -9,34 +39,92 @@
     pub(crate) content_type: ContentType,
     pub(crate) content: MLSPlaintextContentType,
     pub(crate) signature: Signature,
+    pub(crate) confirmation_tag: Option<ConfirmationTag>,
+    pub(crate) membership_tag: Option<MembershipTag>,
 }
 
 impl MLSPlaintext {
-    pub fn new(
-        sender: LeafIndex,
+    /// This constructor builds a new `MLSPlaintext` from the parameters
+    /// provided. It is ony used internally.
+    pub(crate) fn new_from_member(
+        ciphersuite: &Ciphersuite,
+        sender_index: LeafIndex,
         authenticated_data: &[u8],
         content: MLSPlaintextContentType,
         credential_bundle: &CredentialBundle,
         context: &GroupContext,
+        membership_key: &Secret,
     ) -> Self {
-        let sender = Sender {
-            sender_type: SenderType::Member,
-            sender,
-        };
+        let sender = Sender::member(sender_index);
+
         let mut mls_plaintext = MLSPlaintext {
-            group_id: context.group_id.clone(),
-            epoch: context.epoch,
+            group_id: context.group_id().clone(),
+            epoch: context.epoch(),
             sender,
             authenticated_data: authenticated_data.to_vec(),
             content_type: ContentType::from(&content),
             content,
             signature: Signature::new_empty(),
```

Comment:
Well, so the question is whether we want to make our code ugly because the spec says this. I'd strongly object. If the struct in the spec doesn't make sense for our implementation, we either have to make sure that the encoding/decoding accounts for the differences, or have a spec compliant struct and a "usable" struct. (I'd go for the former approach.)
But since this didn't change here, please file a follow-up issue.

Reply:
I have to say, I agree with @franziskuskiefer here. There are quite a few places, where our data structures model the spec 1:1 at the cost of code complexity and I don't think that's a good trade off in most cases. Does the Signature being an `Option` type really make it that much less auditable?

Reply:
Sure, we can agree to be less strict in the implementation in general. In this particular case I'm not really sure what the best solution is. If we make this an optional field, it will only be set to `None` for a very short time when the struct is created. In all other instances we would have to use `unwrap()`, which is ugly and also doesn't make it much safer. I guess a better way would be to rewrite the creation part of the struct entirely.